src/sentry/models/organizationmember.py

```diff
@@ -69,11 +68,9 @@ def get_contactable_members_for_org(self, organization_id: int) -> QuerySet:
 
     def delete_expired(self, threshold: int) -> None:
         """Delete un-accepted member invitations that expired `threshold` days ago."""
+        from sentry.services.hybrid_cloud.authprovider import auth_provider_service
 
-        orgs_with_scim = AuthProvider.objects.filter(
-            flags=F("flags").bitor(AuthProvider.flags.scim_enabled)
-        ).values_list("organization_id", flat=True)
-
+        orgs_with_scim = auth_provider_service.get_org_ids_with_scim()
```

Comment:
This is interesting logic. Getting all orgs from all regions isn't something that is going to work longer term.

Reply:
Agreed. I was thinking maybe we need to denormalize scim_enabled onto organization at some point?

Reply:
This method probably needs to change more dramatically. It is used from the retention cleanup tools which will need to run in each region. Cleanup in one region shouldn't be reaching across silos. Instead we would be better off if the member invitation knew that it was generated by scim, or like you suggested organizations had more context on their scim setup.